### DISCUSSION TITLE
Using your own model failed with Method Spatie\MediaLibrary\MediaCollections\Models\Collections\MediaCollection::logs does not exist.

### DISCUSSION THREAD

**Discussion by nurhazbiy** [spatie/laravel-medialibrary#3307] [Q&A]
I need to extend the files because i was tracking user access of media, so my medias have one to many relationship with logs (lets say MediaLog Collection). but when i use my own model, they still return default model. here is my code

download logs
```
$logs = $course->getMedia('course')->logs();
```

media.php
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Relations\HasMany;
use Spatie\MediaLibrary\MediaCollections\Models\Media as BaseMedia;

class Media extends BaseMedia
{
    /**
     * Get the logs for the media.
     */
    public function logs(): HasMany
    {
        return $this->hasMany(MediaLog::class);
    }
}

```
media-library.php
```
     /*
     * The fully qualified class name of the media model.
     */
    // 'media_model' => Spatie\MediaLibrary\MediaCollections\Models\Media::class,
    'media_model' => App\Models\Media::class,
```

Error 
![image](https://github.com/spatie/laravel-medialibrary/assets/34585999/0cb1a8d1-b04d-4834-a8d9-9c9f8944906d)

**Comment by cosmastech** [comment]
That looks like you're trying to call a method `logs()` on the MediaCollection, but you defined it on the Media model. Is that possible that you need to macro/extend the MediaCollection to return all logs for all media?

**Comment by nurhazbiy** [comment]
yes, i need a custom function, and from what i understand using custom model will return collection from MY media model, not the default media collection. is that not right?

**Comment by vFire** [comment]
Quit simple,
($course->getMedia('course'))->logs();
Don't directly chain the method, just add it into() then call your extended method. that's it.
